### PR TITLE
docs(backends): add operation support matrix

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds an operation support matrix table the backend docs, showing for each backend whether an operation is implemented by the backend:

image

### CODE REVIEW

File: ibis/backends/datafusion/__init__.py [hunk 1]
```diff
@@ -195,3 +196,14 @@ def compile(
         **kwargs: Any,
     ):
         return translate(expr)
+
+    @classmethod
+    def has_operation(self, operation: type[ops.ValueOp]) -> bool:
```
Comment (cpcloud) [inline ibis/backends/datafusion/__init__.py:201]: Should be called `cls`

File: gen_matrix.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    print("---", file=f)
+    print(file=f)
+    print("# Operation Support Matrix", file=f)
+    print(file=f)
+    print(
+        '!!! note "Backends are shown in descending order of the number of supported operations"',  # noqa: E501
+        file=f,
+    )
+    print(file=f)
+    print('<div class="support-matrix" markdown>', file=f)
```
Comment (kszucs) [inline gen_matrix.py:46]: I wonder, could we [feed an existing plugin](https://github.com/timvink/mkdocs-table-reader-plugin) with a csv instead?

Comment (cpcloud) [reply gen_matrix.py:46]: Great idea, will look into it!

Comment (cpcloud) [reply gen_matrix.py]: This worked out pretty nicely. I'm now using the plugin to read the file.

File: ibis/backends/base/sql/__init__.py [hunk 3]
```diff
@@ -252,3 +252,9 @@ def explain(
         cur.release()
 
         return '\n'.join(['Query:', util.indent(query, 2), '', *result])
+
+    @classmethod
+    def has_operation(cls, operation: type[ops.ValueOp]) -> bool:
+        translator = cls.compiler.translator_class
+        op_classes = translator._registry.keys() | translator._rewrites.keys()
```
Comment (kszucs) [inline ibis/backends/base/sql/__init__.py:259]: We should have a special case for `ScalarParameter`

Comment (kszucs) [reply ibis/backends/base/sql/__init__.py:259]: Seems like `Pi` is not implemented for any of the backends, shall we remove it?

Comment (kszucs) [reply ibis/backends/base/sql/__init__.py:259]: `SummaryFilter` is implemented for the SQL backends but there are no tests, I assume it's unused.

**Comment by kszucs** [comment]
I rendered the table locally and it looks pretty blurry for me:

image

Shouldn't be a blocker, but definitely deserves a follow-up ticket.

File: ibis/backends/base/sql/__init__.py (hunk 3 above)
Comment (cpcloud) [reply ibis/backends/base/sql/__init__.py:259]: Mind creating those issues?

**Comment by cpcloud** [comment]
@kszucs Mind approving?

**Pull request merged**